flocker/node/agents/test/blockdevicefactory.py

```diff
@@ -179,7 +178,9 @@ def _openstack(**config):
     # case) instead of forcing me to figure out how to upper case things in
     # bash (I already learned a piece of shell syntax today, once is all I can
     # take).
-    region = environ['FLOCKER_FUNCTIONAL_TEST_OPENSTACK_REGION'].upper()
+    region = environ.get('FLOCKER_FUNCTIONAL_TEST_OPENSTACK_REGION')
```

Comment:
These changes should work on Rackspace too.  I'd confine this PR to the basic address-driven heuristic.  Then make fixes as necessary after there's a PistonCloud builder set up.